File: XBOOK/XBOOK.Web/ClientApp/src/app/pages/_shared/models/journalentry/journalentry.model.ts
export interface JournalEntryViewModel {
    id: number;
    entryName: string;
    description: string;
    dateCreate: string;
    objectType: string;
    objectID: number;
    objectName: string;
}

export class DataMap {
    id: number;
    objectName: string;
    objectType: string;
}
export class Data {
    id: number;
    objectName: string;
    objectType: string;
}

export interface CreateRequest {
    id: number;
    entryName: string;
    description: string;
    dateCreate: string;
    objectType: string;
    objectID: number;
    objectName: string;
    detail: Detail[];
}

export interface Detail {
    id: number;
    accNumber: string;
    crspAccNumber: string;
    note: string;
    debit: number;
    credit: number;
}
export class Datatable {
    accountNumber: string;
    accountName: string;
}

File: XBOOK/XBOOK.Data/Model/JournalEntryModel.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace XBOOK.Data.Model
{
    public class JournalEntryModelCreate
    {
        public long Id { get; set; }
        public string EntryName { get; set; }
        public string Description { get; set; }
        public DateTime DateCreate { get; set; }
        public string ObjectType { get; set; }
        public long? ObjectID { get; set; }
        public string ObjectName { get; set; }
        public List<JournalEntryDetailModelCreate> Detail { get; set; }
    }

    public class JournalEntryDetailModelCreate
    {
        public long Id { get; set; }
        public long JournalID { get; set; }
        public string accNumber { get; set; }
        public string crspAccNumber { get; set; }
        public string note { get; set; }
        public decimal debit { get; set; }
        public decimal credit { get; set; }
    }
}
